src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/ReferenceCountedDisposable.cs

```diff
@@ -3,7 +3,6 @@
 // See the LICENSE file in the project root for more information.
```

Comment:
should this be documented in any of these types?  specifically that no one should ever add anything that would change this as consumption code presumes torn writes will not happen?

Reply:
I would strongly encourage rather than documentation, adding tests to actually catch that. We should be able to write some tests that use reflection to either assert there is a single field, or sizeof(the objects) is not larger than a pointer, etc.

Reply:
I like that idea. It can even just be in the static constructor of we want.